plugin/src/main/java/com/google/errorprone/xplat/checker/JodaTimeLocal.java

```diff
@@ -0,0 +1,189 @@
+            + " due to incompatibilities. A fix using a new DateTime is suggested.",
+    severity = ERROR)
+public class JodaTimeLocal extends BugChecker implements MethodInvocationTreeMatcher,
+    NewClassTreeMatcher {
+
+  private static final ImmutableMap<String, ImmutableList<String>> CLASS_NAMES =
+      ImmutableMap.of("org.joda.time.LocalDateTime", ImmutableList.of("toDateTime"),
+
+          "org.joda.time.LocalDate",
+          ImmutableList.of("toDateTime", "toDateTimeAtCurrentTime", "toDateTimeAtStartOfDay"),
+
+          "org.joda.time.LocalTime", ImmutableList.of("toDateTimeToday"));
+
+
+  private static final Matcher<ExpressionTree> CONSTRUCTOR_MATCHER =
+      Matchers.anyOf(
+          Matchers.anyOf(
+              CLASS_NAMES.keySet().stream()
+                  .map(
+                      typeName ->
+                          Matchers.constructor()
+                              .forClass(typeName)
+                              .withParameters("org.joda.time.DateTimeZone"))
+                  .collect(toImmutableList())),
+          Matchers.constructor()
+              .forClass("org.joda.time.LocalDateTime")
+              .withParameters("long", "org.joda.time.DateTimeZone"),
+          Matchers.constructor()
+              .forClass("org.joda.time.LocalDateTime")
+              .withParameters("java.lang.Object", "org.joda.time.DateTimeZone"));
+
+  private Matcher<ExpressionTree> methodMatcher(String key) {
+    return Matchers.anyOf(
+        CLASS_NAMES.get(key).stream()
+            .map(
+                methodName ->
+                    Matchers.instanceMethod().onExactClass(key).named(methodName))
+            .collect(toImmutableList()));
+  }
+
+  private Description.Builder message(Tree tree, String arg) {
+    return buildDescription(tree)
+        .setMessage(
+            String.format("The use of %s is banned from cross platform development due to"
+                + " incompatibilities.", arg));
+  }
+
+  private Description messageFix(Tree tree, String arg, int start, int end,
+      String replacement) {
+    return message(tree, arg)
+        .addFix(
+            SuggestedFix.builder()
+                .addImport("org.joda.time.DateTimeZone")
+                .replace(
+                    start,
+                    end,
+                    replacement)
+                .build())
+        .build();
+  }
+
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    ExpressionTree recv = ASTHelpers.getReceiver(tree);
+    String recvSrc = state.getSourceForNode(recv);
+    Symbol symbol = ASTHelpers.getSymbol(tree);
+    List<? extends ExpressionTree> arguments = tree.getArguments();
+
+    String argument;
+    if (arguments.isEmpty()) {
+      argument = "DateTimeZone.getDefault()";
+    } else {
+      argument = state.getSourceForNode(arguments.get(0));
+    }
+
+    if (symbol != null && recvSrc != null && recv != null) {
```

Comment:
Readability comment: since the remainder of the method has a very deep true branch and a shallow false branch, consider bail early: `if (symbol == null || recvSrc == null || recv == null) { return Description.NO_MATCH; }`. This way you can flatten this true branch by one level.

Reply:
Done